### ISSUE TITLE
Cannot convert `BigFloat` to `float16`

### ISSUE THREAD

**Issue by eschnett** [JuliaLang/julia#9839]
```
julia> float16(big(3.0))
ERROR: MethodError: `convert` has no method matching convert(::Type{Float16}, ::Base.MPFR.BigFloat)
This may have arisen from a call to the constructor Float16(...), since type constructors fall back to convert methods in julia v0.4.
Closest candidates are:
  convert(::Type{T}, ::T)
  convert(::Type{Bool}, ::Base.MPFR.BigFloat)
  convert(::Type{Bool}, ::Real)
  ...

 in float16 at float.jl:107

julia> float16(float32(big(3.0)))
float16(3.0)
```

**Comment by eschnett** [comment]
Implementation proposed in #9844.

**Comment by ivarne** [comment]
Next time you can add fix #9839 in the PR description,  so that the issue is closed automatically.
